### ISSUE TITLE
Cmake versions higher that 3.14.3 cannot find packages 

### ISSUE THREAD

**Issue by anbenali** [QMCPACK/qmcpack#1574]
with cmake versions equal and newer explicitely than 3.14.3 there is a change in policy that makes it hard to find libraries like HDF5.
```
[benali@hopcroft build_real]$ ../../Libraries/cmake-3.14.3/bin/cmake --help-policy CMP0074
CMP0074
-------

``find_package()`` uses ``<PackageName>_ROOT`` variables.

In CMake 3.12 and above the ``find_package(<PackageName>)`` command now
searches prefixes specified by the ``<PackageName>_ROOT`` CMake
variable and the ``<PackageName>_ROOT`` environment variable.
Package roots are maintained as a stack so nested calls to all ``find_*``
commands inside find modules also search the roots as prefixes.  This policy
provides compatibility with projects that have not been updated to avoid using
``<PackageName>_ROOT`` variables for other purposes.

The ``OLD`` behavior for this policy is to ignore ``<PackageName>_ROOT``
variables.  The ``NEW`` behavior for this policy is to use
``<PackageName>_ROOT`` variables.

This policy was introduced in CMake version 3.12.  CMake version
3.14.3 warns when the policy is not set and uses ``OLD`` behavior.
Use the ``cmake_policy()`` command to set it to ``OLD`` or ``NEW``
explicitly.

.. note::
  The ``OLD`` behavior of a policy is
  ``deprecated by definition``
  and may be removed in a future version of CMake.
```

The Cmake output gives this warning:
```
-- Found LibXml2: /lib64/libxml2.so (found version "2.9.1") 
CMake Warning (dev) at CMakeLists.txt:604 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable HDF5_ROOT is set to:

    /home/benali/src/Libraries/hdf5-1.10.5/Libraries

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: hdf5;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so  found components:  C 
-- Using HDF5 non-scalable serial I/O due to the lack of library support for parallel
```
==============
And the code dies at compilation with:
```
/home/benali/src/qmcpack/src/io/hdf_datatype.h:18:10: fatal error: hdf5.h: No such file or directory
 #include <hdf5.h>
          ^~~~~~~~
```
because neither include path nor linking path are added properly.

**Comment by prckent** [comment]
Please try setting the cmake_policy in the CMakeLists.txt. If this works we can onboard "old" usage for now.

**Comment by markdewing** [comment]
I'm a little confused - the FindHDF module explicitly checks HDF5_ROOT.   How does the policy introduction affect this?

**Comment by ye-luo** [comment]
````
CMake version 3.14.3 warns when the policy is not set and uses ``OLD`` behavior.
```
```
The ``OLD`` behavior for this policy is to ignore ``<PackageName>_ROOT``
variables. 
```
Basically HDF5_ROOT is ignored by cmake. It seems that the “OLD” behavior is not really the old behavior.

**Comment by markdewing** [comment]
I think the message for the policy refers to the CMake core, not the individual modules.   With the change (the NEW) policy, the CMake find_package command automatically adds the `<pkg>_ROOT` variable to the list of places to look.   With the OLD policy, the find_package command does not.  
The individual 'find' modules can do whatever they want.  In the case of FindHDF5, it uses HDF5_ROOT.

I suspect this change was made so all the modules have a uniform way to specify a package root, and not have to implement the same code in every module.

**Comment by prckent** [comment]
Closed by #1705
